bin/generate_practice_exercise.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+# shellcheck source=/dev/null
+source ./bin/generator-utils/utils.sh
+source ./bin/generator-utils/prompts.sh
+source ./bin/generator-utils/templates.sh
+
+# Exit if anything fails.
+set -euo pipefail
+
+# If argument not provided, print usage and exit
+if [ $# -ne 1 ]; then
+    echo "Usage: bin/generate_practice_exercise.sh <exercise-slug>"
+    exit 1
+fi
```

Comment:
It would be nice if you could run this non-interactively via:

`bin/generate_practice_exercise.sh <exercise-slug> [difficulty] [author-github-handle]`

The last two arguments being optional.